### PR TITLE
PMM-6585 Always expose mongodb_up metric

### PR DESCRIPTION
**Pull request by percona-csalguero**
New PR to replace https://github.com/percona/mongodb_exporter/pull/207
This is based on PMM-6610

### CODE REVIEW

File: exporter/general_collector.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	var value float64
+
+	if err := client.Ping(ctx, readpref.PrimaryPreferred()); err == nil {
+		value = 1
+	}
+
+	d := prometheus.NewDesc("mongodb_up", "Whether MongoDB is up.", nil, nil)
+	up, err := prometheus.NewConstMetric(d, prometheus.GaugeValue, value)
+	if err != nil {
+		panic(err)
```
Comment (BupycHuk) [inline exporter/general_collector.go:52]: We should not panic

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/general_collector.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+func (d *generalCollector) Collect(ch chan<- prometheus.Metric) {
+	ch <- mongodbUpMetric(d.ctx, d.client)
+}
+
+func mongodbUpMetric(ctx context.Context, client *mongo.Client) prometheus.Metric {
+	var value float64
+
+	if err := client.Ping(ctx, readpref.PrimaryPreferred()); err == nil {
```
Comment (BupycHuk) [inline exporter/general_collector.go:45]: let's log error.

**Review by BupycHuk — CHANGES_REQUESTED** [review]

File: exporter/general_collector.go (hunk 2 above)
Comment (BupycHuk) [reply exporter/general_collector.go:47]: @percona-csalguero it's not addressed yet.

Comment (percona-csalguero) [reply exporter/general_collector.go:47]: done

Comment (BupycHuk) [reply exporter/general_collector.go:47]: you are logging nil err

Comment (percona-csalguero) [reply exporter/general_collector.go]: sorry .

File: exporter/general_collector.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+func mongodbUpMetric(ctx context.Context, client *mongo.Client, log *logrus.Logger) prometheus.Metric {
+	var value float64
+
+	err := client.Ping(ctx, readpref.PrimaryPreferred())
+	if err == nil {
+		value = 1
+	}
+	log.Errorf("error while checking mongodb connection: %s", err)
```
Comment (BupycHuk) [inline exporter/general_collector.go:51]: now you are logging even if there no error.

**Review by BupycHuk — APPROVED** [review]

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: exporter/general_collector.go (hunk 1 above)
Comment (AlekSi) [reply exporter/general_collector.go]: @BupycHuk @percona-csalguero It is totally fine to panic on programmer's mistakes (as opposed to environment) to make them visible. That's why we use `MustRegister`, and that's why prometheus package provides `MustNewConstMetric` that should be used there.

**Review by AlekSi — APPROVED** [review]
LGTM, please merge once CI passes.

But please also use `MustNewConstMetric` where it does not depend on environment (if MongoDB down, if version is incorrect, etc).

File: exporter/general_collector.go (hunk 1 above)
Comment (percona-csalguero) [reply exporter/general_collector.go]: :+1:

**Pull request merged**